lib/Logger.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * @package ActiveRecord
+ */
+namespace ActiveRecord;
+
+/**
+ * Defines a logger for logging queries.
+ *
+ * @package ActiveRecord
+ */
+interface Logger
+{
+
+	/**
+	 * Logs a message.
+	 *
+	 * Implementations should expect that `done()` will be called after a call
+	 * to this method, for logged SQL queries.
+	 *
+	 * @param string $message The message to log.
+	 * @param array $params Any parameters associated with the message.
+	 * @return void
+	 */
+	public function log($message, array $params = null);
```

Comment:
Kind of a nit, but we may want to copy Doctrine's naming to make things a little more explicit/clear.

`log` usually implies a generic logging method, however this is specific to logging a query. Same for `done`, which might be a little unclear about when that happens. 

Doctrine SQL Logger interface is very clear:

https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Logging/SQLLogger.php

```php
interface SQLLogger
{
    /**
     * Logs a SQL statement somewhere.
     *
     * @param string     $sql    The SQL to be executed.
     * @param array|null $params The SQL parameters.
     * @param array|null $types  The SQL parameter types.
     *
     * @return void
     */
    public function startQuery($sql, array $params = null, array $types = null);
    /**
     * Marks the last started query as stopped. This can be used for timing of queries.
     *
     * @return void
     */
    public function stopQuery();
}
```